test/gtest/uct/ib/test_devx.cc

```diff
@@ -49,3 +49,235 @@ UCS_TEST_P(test_devx, dbrec)
+    /* UMR mkey pool must be empty */
+    ASSERT_TRUE(ucs_list_is_empty(&md()->umr.mkey_pool));
+    /* Whether mkey is UMR depends on XGVMI capability */
+    check_memh_export(memh, true);
+
+    destroy_memh(memh);
+
+    /* After release of memh, associated UMR mkey is moved back to mkey pool */
+    ASSERT_EQ(1, ucs_list_length(&md()->umr.mkey_pool));
+}
+
+UCS_TEST_P(test_devx_umr_mkey, import_mkey_no_umr, "IB_XGVMI_UMR_ENABLE=n")
+{
+    if (!check_xgvmi()) {
+        UCS_TEST_SKIP_R("XGVMI capability is not supported");
+    }
+
+    /* Export any mkey */
+    uct_ib_mlx5_devx_mem_t *exported_memh = create_memh(1024);
+    ASSERT_UCS_OK(uct_ib_mlx5_devx_reg_exported_key(md(), exported_memh));
+
+    /* Import as non-UMR mkey */
+    uct_ib_mlx5_devx_mem_t *imported_memh = import_memh(exported_memh);
+    ASSERT_NE(nullptr, imported_memh);
+
+    /* UMR hash map remains uninitialized */
+    ASSERT_EQ(nullptr, md()->umr.mkey_hash);
+
+    destroy_memh(imported_memh);
+    destroy_memh(exported_memh);
+}
+
+UCS_TEST_P(test_devx_umr_mkey, import_mkey_umr)
+{
+    if (!check_xgvmi()) {
+        UCS_TEST_SKIP_R("XGVMI capability is not supported");
+    }
+
+    /* Export UMR mkey */
+    uct_ib_mlx5_devx_mem_t *exported_memh = create_memh(1024);
+    ASSERT_UCS_OK(uct_ib_mlx5_devx_reg_exported_key(md(), exported_memh));
+    check_memh_export(exported_memh, true);
+
+    /* Import first UMR mkey */
+    uct_ib_mlx5_devx_mem_t *imported_memh = import_memh(exported_memh);
+
+    check_umr_init(true, true);
+    ASSERT_EQ(1, kh_size(md()->umr.mkey_hash));
+
+    /* Second import with the same packed mkey, should reuse alias */
+    uct_ib_mlx5_devx_mem_t *imported_memh2 = import_memh(exported_memh);
+    ASSERT_NE(imported_memh, imported_memh2);
+    ASSERT_EQ(imported_memh->super.lkey, imported_memh2->super.lkey);
+    ASSERT_EQ(1, kh_size(md()->umr.mkey_hash));
+
+    destroy_memh(imported_memh);
+    destroy_memh(imported_memh2);
+    destroy_memh(exported_memh);
+}
+
+UCS_TEST_P(test_devx_umr_mkey, perf_export_import_mkey_xgvmi)
+{
+    if (!check_xgvmi()) {
+        UCS_TEST_SKIP_R("XGVMI capability is not supported");
+    }
+
+    const size_t MEMH_COUNT = 1000;
+    ucs_time_t export_create_time, import_create_time;
+    perf_export_import_mkey(MEMH_COUNT, 1024, export_create_time,
+                            import_create_time);
+
+    /* After release all allocated UMR keys go back to the mkey pool */
+    ASSERT_EQ(MEMH_COUNT, ucs_list_length(&md()->umr.mkey_pool));
+
+    /* Now reuse the keys for different handles of different size */
+    ucs_time_t export_reuse_time, import_reuse_time;
+    perf_export_import_mkey(MEMH_COUNT, 2048, export_reuse_time,
+                            import_reuse_time);
+
+    /* Reusing existing UMR mkeys is normally 200x faster than creating new
+     * ones, given jitter we check that it's at least 50 times faster */
+    ASSERT_GT(export_create_time / 50, export_reuse_time);
```

Comment:
1. IMO we can reduce it to 10 times, or check vs. some known time content
2. inverse order since export_reuse_time is what we test:
EXPECT_LT(export_reuse_time, export_create_time / 10)